libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -139,9 +182,55 @@ void DocStringAnalyser::handleCallable(
 	StructurallyDocumentedAnnotation& _annotation
 )
 {
-	static set<string> const validTags = set<string>{"author", "dev", "notice", "return", "param"};
+	static set<string> const validTags = set<string>{"author", "dev", "notice", "return", "param", "inheritdoc"};
 	parseDocStrings(_node, _annotation, validTags, "functions");
 	checkParameters(_callable, _node, _annotation);
+
+	if (CallableDeclaration const* baseFunction = resolveInheritDoc(_callable, _node, _annotation))
+		copyMissingTags(_annotation, baseFunction->annotation());
+	else if (
+		_annotation.docTags.empty() &&
+		_callable.annotation().baseFunctions.size() == 1 &&
+		parameterNamesEqual(_callable, **_callable.annotation().baseFunctions.begin())
+	)
+		copyMissingTags(_annotation, (*_callable.annotation().baseFunctions.begin())->annotation());
+}
+
+CallableDeclaration const* DocStringAnalyser::resolveInheritDoc(
+	CallableDeclaration const& _callable,
+	StructurallyDocumented const& _node,
+	StructurallyDocumentedAnnotation& _annotation
+)
+{
+	size_t const occurrences = _annotation.docTags.count("inheritdoc");
+
+	if (occurrences == 0)
+		return nullptr;
+
+	if (occurrences > 1)
+	{
+		m_errorReporter.docstringParsingError(
+			5142_error,
+			_node.documentation()->location(),
+			"Documentation tag @inheritdoc can only reference one contract."
+		);
+		return nullptr;
+	}
+
+	string const targetContractName = _annotation.docTags.find("inheritdoc")->second.content;
+
+	if (auto callable = findBaseCallable(_callable, targetContractName))
+		return callable;
+
+	m_errorReporter.docstringParsingError(
+		1430_error,
+		_node.documentation()->location(),
+		"Documentation tag @inheritdoc references contract \"" +
+		targetContractName +
+		"\", but the contract doesn't override this function."
```

Comment:
Also when the contract doesn't exist, right?

Reply:
That is true, but finding out which case it is is a bit more work which I deemed not worth the extra logic just for a slightly more detailed error message. We could rephrase it to mean both though.

Reply:
I think tweaking the message is fine.